Subject: [PATCH] [BZ 184396] zipl: fix zipl.conf man page example for secure boot
From: Stefan Haberland <sth@linux.ibm.com>

Description:   zipl: fix secure boot config handling
Symptom:       The config file parsing for secure boot worked not as
               it was expected to be. For example a config section
               setting was not evaluated properly.
               It is not possible to specify command line option -S
               without other options.
               Additionally the man page showed an invalid example.
Problem:       The config file parsing was not implemented properly.
Solution:      The hierarchy of the secure boot settings in the config
               file is:
                    defaultboot > menu > section
               Allow that --secure or -S is specified on command line
               without the need to allow all options on the command
               line. Also ensure that the command line option
               overrules the config option and correctly ensure that
               secure boot is only set for SCSI devices.
               Fix man page example.
Reproduction:  Run zipl with a secure= setting in a configuration
               section or specify -S on command line.
Upstream-ID:   299fd2b7729f35c6fe3be18964f7e5e6a365f94d
Problem-ID:    184396

Upstream-Description:

              zipl: fix zipl.conf man page example for secure boot

              The secure= option is not supported in the defaultboot section when a
              menu is used. It should be placed in the menu section in this case.

              Signed-off-by: Stefan Haberland <sth@linux.ibm.com>
              Signed-off-by: Jan Hoeppner <hoeppner@linux.ibm.com>


Signed-off-by: Stefan Haberland <sth@linux.ibm.com>
---
 zipl/man/zipl.conf.5 |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/zipl/man/zipl.conf.5
+++ b/zipl/man/zipl.conf.5
@@ -82,8 +82,6 @@ below).
 .br
 defaultmenu = menu1
 .br
-secure = auto
-.br
 
 [linux]
 .br
@@ -117,6 +115,8 @@ prompt      = 1
 .br
 timeout     = 0
 .br
+secure      = auto
+.br
 .PP
 
 .B BootLoaderSpec configuration files
@@ -522,7 +522,7 @@ non-default memory location.
 .B secure
 =
 .IR auto / 1 / 0
-(configuration only)
+(configuration and menu)
 .IP
 .B Configuration section:
 .br
